spec/models/canonical/armor_spec.rb

```diff
@@ -189,29 +189,44 @@
       end
     end
 
-    describe 'smithing materials' do
+    describe 'crafting materials' do
       let(:armor) { create(:canonical_armor) }
-      let(:material) { create(:canonical_raw_material) }
 
       before do
-        armor.canonical_craftables_crafting_materials.create!(material:, quantity: 4)
+        armor.canonical_crafting_materials.create!(
+          quantity: 4,
+          source_material: create(:canonical_raw_material),
+        )
+        armor.reload
       end
 
       it 'gives the quantity needed' do
-        expect(armor.crafting_materials.first.quantity_needed).to eq 4
+        material = armor
+                     .crafting_materials
+                     .first
+                     .materials
+                     .first
+
+        expect(material.quantity).to eq 4
       end
     end
 
     describe 'tempering materials' do
       let(:armor) { create(:canonical_armor) }
-      let(:material) { create(:canonical_raw_material) }
 
       before do
-        armor.canonical_temperables_tempering_materials.create!(material:, quantity: 1)
+        create(
+          :canonical_material,
+          temperable: armor,
+          quantity: 1,
+        )
+
+        armor.reload
       end
 
       it 'gives the quantity needed' do
```

Comment:
As above so below 😅 This test also needs to be updated to test something meaningful.